File: Archiver/app_code/clsAVAILFILETYPESUNDEFINED.cs
using global::System.Data.SqlClient;
using System.Windows.Forms;
using global::Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public class clsAVAILFILETYPESUNDEFINED : clsDatabaseARCH
    {
        // ** DIM the selected table columns 
        // Dim DBARCH As New clsDatabaseARCH
        private clsDma DMA = new clsDma();
        private clsUtility UTIL = new clsUtility();
        private clsLogging LOG = new clsLogging();
        private string FileType = "";
        private string SubstituteType = "";
        private string Applied = "";


        // ** Generate the SET methods 
        public void setFiletype(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Filetype' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            FileType = val;
        }

        public void setSubstitutetype(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            SubstituteType = val;
        }

        public void setApplied(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                val = "null";
            }

            val = UTIL.RemoveSingleQuotes(val);
            Applied = val;
        }



        // ** Generate the GET methods 
        public string getFiletype()
        {
            if (Strings.Len(FileType) == 0)
            {
                MessageBox.Show("GET: Field 'Filetype' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(FileType);
        }

        public string getSubstitutetype()
        {
            return UTIL.RemoveSingleQuotes(SubstituteType);
        }

        public string getApplied()
        {
            if (Strings.Len(Applied) == 0)
            {
                Applied = "null";
            }

            return Applied;
        }



        // ** Generate the Required Fields Validation method 
        public bool ValidateReqData()
        {
            if (FileType.Length == 0)
                return false;
            return true;
        }


        // ** Generate the Validation method 
        public bool ValidateData()
        {
            if (FileType.Length == 0)
                return false;
            return true;
        }


        // ** Generate the INSERT method 
        public bool Insert()
        {
            bool b = false;
            string s = "";
            s = s + " INSERT INTO AvailFileTypesUndefined(";
            s = s + "FileType,";
            s = s + "SubstituteType,";
            s = s + "Applied) values (";
            s = s + "'" + FileType + "'" + ",";
            s = s + "'" + SubstituteType + "'" + ",";
            s = s + Applied + ")";
            return ExecuteSqlNewConn(s, false);
        }


        // ** Generate the UPDATE method 
        public bool Update(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            s = s + " update AvailFileTypesUndefined set ";
            s = s + "FileType = '" + getFiletype() + "'" + ", ";
            s = s + "SubstituteType = '" + getSubstitutetype() + "'" + ", ";
            s = s + "Applied = " + getApplied();
            WhereClause = " " + WhereClause;
            s = s + WhereClause;
            return ExecuteSqlNewConn(s, false);
        }


        // ** Generate the SELECT method 
        public SqlDataReader SelectRecs()
        {
            bool b = false;
            string s = "";
            SqlDataReader rsData;
            s = s + " SELECT ";
            s = s + "FileType,";
            s = s + "SubstituteType,";
            s = s + "Applied ";
            s = s + " FROM AvailFileTypesUndefined";
            string CS = setConnStr();
            var CONN = new SqlConnection(CS);
            CONN.Open();
            var command = new SqlCommand(s, CONN);
            rsData = command.ExecuteReader();
            return rsData;
        }


        // ** Generate the Select One Row method 
        public SqlDataReader SelectOne(string WhereClause)
        {
            bool b = false;
            string s = "";
            SqlDataReader rsData;
            s = s + " SELECT ";
            s = s + "FileType,";
            s = s + "SubstituteType,";
            s = s + "Applied ";
            s = s + " FROM AvailFileTypesUndefined";
            s = s + WhereClause;
            string CS = setConnStr();
            var CONN = new SqlConnection(CS);
            CONN.Open();
            var command = new SqlCommand(s, CONN);
            rsData = command.ExecuteReader();
            return rsData;
        }


        // ** Generate the DELETE method 
        public bool Delete(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            WhereClause = " " + WhereClause;
            s = " Delete from AvailFileTypesUndefined";
            s = s + WhereClause;
            b = ExecuteSqlNewConn(s, false);
            return b;
        }


        // ** Generate the Zeroize Table method 
        public bool Zeroize()
        {
            bool b = false;
            string s = "";
            s = s + " Delete from AvailFileTypesUndefined";
            b = ExecuteSqlNewConn(s, false);
            return b;
        }


        // ** Generate Index Queries 
        public int cnt_PK_AFTU(string FileType)
        {
            int B = 0;
            string TBL = "AvailFileTypesUndefined";
            string WC = "Where FileType = '" + FileType + "'";
            B = iGetRowCount(TBL, WC);
            return B;
        }     // ** cnt_PK_AFTU

        // ** Generate Index ROW Queries 
        public SqlDataReader getRow_PK_AFTU(string FileType)
        {
            SqlDataReader rsData = null;
            string TBL = "AvailFileTypesUndefined";
            string WC = "Where FileType = '" + FileType + "'";
            rsData = GetRowByKey(TBL, WC);
            if (rsData.HasRows)
            {
                return rsData;
            }
            else
            {
                return null;
            }
        }     // ** getRow_PK_AFTU

        /// Build Index Where Caluses
        public string wc_PK_AFTU(string FileType)
        {
            string WC = "Where FileType = '" + FileType + "'";
            return WC;
        }     // ** wc_PK_AFTU
    }
}

File: Archiver/app_code/clsAVAILFILETYPESUNDEFINED.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql
 
Public Class clsAVAILFILETYPESUNDEFINED
    Inherits clsDatabaseARCH
    '** DIM the selected table columns 
    'Dim DBARCH As New clsDatabaseARCH
    Dim DMA As New clsDma
    Dim UTIL As New clsUtility
    Dim LOG As New clsLogging


    Dim FileType As String = ""
    Dim SubstituteType As String = ""
    Dim Applied As String = ""


    '** Generate the SET methods 
    Public Sub setFiletype(ByRef val AS String )
        If Len(val) = 0 Then
            messagebox.show("SET: Field 'Filetype' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        FileType = val
    End Sub

    Public Sub setSubstitutetype(ByRef val AS String )
        val = UTIL.RemoveSingleQuotes(val)
        SubstituteType = val
    End Sub

    Public Sub setApplied(ByRef val AS String )
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        Applied = val
    End Sub



    '** Generate the GET methods 
    Public Function getFiletype() As String
        If Len(FileType) = 0 Then
            messagebox.show("GET: Field 'Filetype' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(FileType)
    End Function

    Public Function getSubstitutetype() As String
        Return UTIL.RemoveSingleQuotes(SubstituteType)
    End Function

    Public Function getApplied() As String
        If Len(Applied) = 0 Then
            Applied = "null"
        End If
        Return Applied
    End Function



    '** Generate the Required Fields Validation method 
    Public Function ValidateReqData() As Boolean
        If FileType.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the Validation method 
    Public Function ValidateData() As Boolean
        If FileType.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the INSERT method 
    Public Function Insert() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""
        s = s + " INSERT INTO AvailFileTypesUndefined("
        s = s + "FileType,"
        s = s + "SubstituteType,"
        s = s + "Applied) values ("
        s = s + "'" + FileType + "'" + ","
        s = s + "'" + SubstituteType + "'" + ","
        s = s + Applied + ")"
        Return ExecuteSqlNewConn(s, False)

    End Function


    '** Generate the UPDATE method 
    Public Function Update(ByVal WhereClause AS String ) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        s = s + " update AvailFileTypesUndefined set "
        s = s + "FileType = '" + getFiletype() + "'" + ", "
        s = s + "SubstituteType = '" + getSubstitutetype() + "'" + ", "
        s = s + "Applied = " + getApplied()
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return ExecuteSqlNewConn(s, False)
    End Function


    '** Generate the SELECT method 
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "FileType,"
        s = s + "SubstituteType,"
        s = s + "Applied "
        s = s + " FROM AvailFileTypesUndefined"

        Dim CS As String = setConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader() 
        Return rsData
    End Function


    '** Generate the Select One Row method 
    Public Function SelectOne(ByVal WhereClause AS String ) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "FileType,"
        s = s + "SubstituteType,"
        s = s + "Applied "
        s = s + " FROM AvailFileTypesUndefined"
        s = s + WhereClause

        Dim CS As String = setConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader() 
        Return rsData
    End Function


    '** Generate the DELETE method 
    Public Function Delete(ByVal WhereClause AS String ) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from AvailFileTypesUndefined"
        s = s + WhereClause

        b = ExecuteSqlNewConn(s, False)
        Return b

    End Function


    '** Generate the Zeroize Table method 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from AvailFileTypesUndefined"

        b = ExecuteSqlNewConn(s, False)
        Return b

    End Function


    '** Generate Index Queries 
    Public Function cnt_PK_AFTU(ByVal FileType As String) As Integer

        Dim B As Integer = 0
Dim TBL AS String  = "AvailFileTypesUndefined" 
Dim WC AS String  = "Where FileType = '" + FileType + "'"

        B = iGetRowCount(TBL, WC)

        Return B
    End Function     '** cnt_PK_AFTU

    '** Generate Index ROW Queries 
    Public Function getRow_PK_AFTU(ByVal FileType As String) As SqlDataReader

        Dim rsData As SqlDataReader = Nothing
Dim TBL AS String  = "AvailFileTypesUndefined" 
Dim WC AS String  = "Where FileType = '" + FileType + "'"

        rsData = GetRowByKey(TBL, WC)

        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK_AFTU

    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK_AFTU(ByVal FileType As String) As String

Dim WC AS String  = "Where FileType = '" + FileType + "'" 

        Return WC
    End Function     '** wc_PK_AFTU
End Class
